packages/utils/node-resolver-core/src/NodeResolver.js

```diff
@@ -273,6 +288,44 @@ export default class NodeResolver {
       // ignore
     }
 
+    // Auto install node builtin polyfills if not already available
+    if (resolved === undefined && builtin != null) {
+      let packageManager = this.packageManager;
+      if (packageManager) {
+        this.logger?.warn({
+          message: md`Auto installing polyfill for Node builtin module "${specifier}"...`,
+          codeFrames: [
+            {
+              filePath: ctx.loc?.filePath ?? sourceFile,
+              codeHighlights: ctx.loc
+                ? [
+                    {
+                      message: 'used here',
+                      start: ctx.loc.start,
+                      end: ctx.loc.end,
+                    },
+                  ]
+                : [],
+            },
+          ],
+          documentationURL:
+            'https://parceljs.org/features/node-emulation/#polyfilling-%26-excluding-builtin-node-modules',
+        });
+
+        await packageManager.resolve(builtin, this.projectRoot + '/index', {
+          saveDev: true,
+          shouldAutoInstall: true,
```

Comment:
Looks like this would override `--no-autoinstall` which should not be the case?

Reply:
The package manager is not passed in as an option from the DefaultResolver if shouldAutoInstall is false